NEWS.md

```diff
@@ -1,3 +1,7 @@
+## 4.2.0 2018-11-23
```

Comment:
Why 4.2.0 but not 4.1.0?

Reply:
there is another PR for this module ([MODUSERBL-50 Implementing reset password link generation]( https://github.com/folio-org/mod-users-bl/pull/50)) and we are expecting that my changes will be merged aftere that PR so I incremented version from that PR